opus/application/static_media/js/browse.js

```diff
@@ -957,19 +958,18 @@ var o_browse = {
             });
 
             galleryHtml += "</div>";
-        }
-
-        // wondering if there should be more logic here to determine if the new block of observations
-        // is contiguous w/the existing block of observations, not just before/after...
-        if ($(`${namespace} .thumbnail-container`).first().data("obs") > data.start_obs) {
-            $(".gallery", namespace).prepend(galleryHtml);
-            if (namespace == "#browse") {   // not yet supported for cart
-                $(".dataTable tbody").prepend(tableHtml);
-            }
-        } else {
-            $(".gallery", namespace).append(galleryHtml);
-            if (namespace == "#browse") {   // not yet supported for cart
-                $(".dataTable tbody").append(tableHtml);
+            // wondering if there should be more logic here to determine if the new block of observations
+            // is contiguous w/the existing block of observations, not just before/after...
+            if ($(`${namespace} .thumbnail-container`).first().data("obs") > data.start_obs) {
```

Comment:
If the gallery is empty, isn't there no first() element? In which case I assume first().data("obs") returns "undefined". That makes the if confusing. Perhaps you should check if there's a first() element at all, and only then do the comparison?